crates/ruff_python_parser/src/lexer.rs

```diff
@@ -339,8 +344,15 @@ impl<'source> Lexer<'source> {
                 let imag = f64::from_str(number.as_str()).unwrap();
                 Ok(Tok::Complex { real: 0.0, imag })
             } else {
-                let value = number.as_str().parse::<BigInt>().unwrap();
-                if start_is_zero && !value.is_zero() {
+                // 2**31 is 2147483648, i.e 10 digits, so all 9 digit numbers are representable as i32
+                // TODO: Can we go up to i64?
+                if number.as_str().len() < 10 {
+                    let value = Integer::from(number.as_str().parse::<i32>().unwrap());
+                    return Ok(Tok::Int { value });
+                }
+
+                let value = Integer::from_str(number.as_str()).unwrap();
+                if start_is_zero && value != 0 {
```

Comment:
Nit: This check is a bit awkward, but you have to make sure your code still passes it